src/Features/Core/Portable/Workspace/CompileTimeSolutionProvider.cs

```diff
@@ -133,10 +135,27 @@ public Solution GetCompileTimeSolution(Solution designTimeSolution)
                 compileTimeSolution = _designTimeToCompileTimeSolution.GetValue(designTimeSolution, _ => compileTimeSolution);
                 _lastCompileTimeSolution = compileTimeSolution;
 
+                if (projectsWithDesignTimeDocuments.Count > 0)
+                {
+                    // HACK: Pre-request all razor source generated documents to warm up the caches
+                    _ = LoadSourceGeneratedDocumentsAsync(compileTimeSolution, projectsWithDesignTimeDocuments.ToImmutableArray(), CancellationToken.None);
```

Comment:
❗ Asynchronous operations must always be tracked, either by the life of the caller (e.g. when awaiting) or through `IAsynchronousOperationListener`